first_ruby/.gitattributes

```diff
@@ -0,0 +1,10 @@
+# See https://git-scm.com/docs/gitattributes for more about git attribute files.
+
+# Mark the database schema as having been generated.
+db/schema.rb linguist-generated
+
+# Mark the yarn lockfile as having been generated.
+yarn.lock linguist-generated
+
+# Mark any vendored files as having been vendored.
+vendor/* linguist-vendored
```

Comment:
What is `linguist-generated` ?
What is this file all about?

Reply:
`linguist-generated` is which the file-path will be excluded from `stats` and hidden in `diffs`. 
`Stats` is the counting of the repository language.
`Diffs` is the displays of differences between files in two commits between a commit and current repository.

`.gitattributes` file is used to specify files and paths attributes when performing git action like `git commit`. One of the common usage of this file is to define how end of line content store in repository in the format of `lf (line feed)` which code in `\n` a new line or `crlf (carriage return line feed)` which code in `\r` moving the cursor to beginning of line. Apparently the way OS handles end of line differently.

Reply:
You mentioned in your other reply that `yarn.lock` is responsible for setting the specific library versions to be used. If yarn.lock is not displayed as diffs and not committed, then how can we always have the same library version?

Reply:
`yarn.lock` is committed but the diffs are hidden. In this case, it will show this for `yarn.lock`. 
![image](https://user-images.githubusercontent.com/80312100/123427264-fc824900-d5f6-11eb-87dd-4e92f7e91d3e.png)